sprocs/question_order.sql

```diff
@@ -14,11 +39,15 @@ SELECT
             END
         WHEN a.type = 'Exam' THEN (row_number() OVER w)::text
         ELSE aq.number::text
-    END AS question_number
+    END AS question_number,
+    NOT ((lag(aq.id) OVER w) IS NULL) -- Don't block if first question in assessment
+        AND BOOL_OR(ln_.locking) OVER (ROWS BETWEEN UNBOUNDED PRECEDING AND 1 PRECEDING) -- Or if the previous question locks ahead
+    AS sequence_locked
 FROM
     assessment_instances AS ai
     JOIN assessments AS a ON (a.id = ai.assessment_id)
     JOIN instance_questions AS iq ON (iq.assessment_instance_id = ai.id)
+    JOIN locks_next AS ln_ ON (ln_.instance_question_id = iq.id)
```

Comment:
Not that I really mind having a trailing underscore, but it's not a pattern used elsewhere in SQL in PL. We could call this variable just `ln`? Or just don't alias it at all and refer to it as `locks_next`. We only reference it once, and it's not a standard PL variable with a standard alias, so an alias isn't really needed anyway.

Reply:
Changed. BTW `ln` is reserved for the natural logarithm function!